contrib/exemplar_util/src/main/java/io/opencensus/contrib/exemplar/util/ExemplarUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.exemplar.util;
+
+import io.opencensus.stats.AggregationData.DistributionData.Exemplar;
+import io.opencensus.stats.MeasureMap;
+import io.opencensus.trace.SpanId;
+import io.opencensus.trace.TraceId;
+import javax.annotation.Nullable;
+
+/**
+ * Utils for encoding and putting special contextual information of {@link Exemplar}s.
+ *
+ * @since 0.16
+ */
+public final class ExemplarUtils {
+
+  /**
+   * Key for {@link TraceId} in the contextual information of an {@link Exemplar}.
+   *
+   * <p>For the {@code TraceId} value of this key, it is suggested to encode it in hex (base 16)
+   * lower case.
+   *
+   * @since 0.16
+   */
+  public static final String ATTACHMENT_KEY_TRACE_ID = "TraceId";
+
+  /**
+   * Key for {@link SpanId} in the contextual information of an {@link Exemplar}.
+   *
+   * <p>For the {@code SpanId} value of this key, it is suggested to encode it in hex (base 16)
+   * lower case.
+   *
+   * @since 0.16
+   */
+  public static final String ATTACHMENT_KEY_SPAN_ID = "SpanId";
+
+  /**
+   * Encodes a {@link TraceId} to a string in base 16 lower case encoding, and puts the encoded
+   * string into the attachments of the given {@link MeasureMap}.
+   *
+   * @param measureMap the {@code MeasureMap}
+   * @param traceId the {@code TraceId} to be encoded and put.
+   * @since 0.16
+   */
+  public static void putTraceIdAsAttachment(MeasureMap measureMap, TraceId traceId) {
```

Comment:
It might be more useful to only provide one method that takes a MeasureMap and a Span or SpanContext and adds both attachments.  We could extend it later to add other fields from the span without users needing to modify their code.

Reply:
I would prefer to keep them separate because there are cases when only `TraceId` is needed, without `SpanId`. From [Stackdriver documentation](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#exemplar):
> Trace ID: type.googleapis.com/google.devtools.cloudtrace.v1.Trace

Reply:
I'm not sure I understand, though I'm not very familiar with exemplars.  I thought that the Stackdriver documentation just used trace ID as an example of an attachment.  What is the use case for only adding a trace ID?  Is that the most common use case?  I had thought that the main use case was adding information about the span, which could explain something like  the high latency of a request within a trace.

/cc @bogdandrutu

Reply:
I would prefer one method that takes the SpanContext. we can do different optimizations in the future if needed (record only if sampled), also I think it is easier for users.

Reply:
SG, done.

> What is the use case for only adding a trace ID? Is that the most common use case?

Stackdriver backend will perform check on the exemplar attachments, and trace id alone (with no span id) is one permitted use case, though I'm not sure how they make use of it in the backend. Anyway I'm fine with starting with SpanContext.